feat: add seamlyme welcome screen

DSCaskey opened the pull request:
This adds a Welcome screen dialog in place of the existing Splash screen. Offers more options and control than a QSplashScreen does. 

*  Allows for form layout with more than a single message.  The existing splash screen was just a background pixmap, which lacks the ability to translate. 
*  Has an OK button that is user dependent and does not rely on a timed delay.
*  Allows for the user to choose whether to see the dialog again.
*  Adds some default startup user prefs such as the Decimal separator that has been an issue for some new users when not set correctly. CheckBox text changes from "Use OS options" to "User locale"
*  Specifically addresses the 3D Look body scan conversion. Also removes the code to attempt creating a conversion email. 

![welcome](https://github.com/FashionFreedom/Seamly2D/assets/31944718/82dcf1d7-b61c-4f11-82c1-a345f6a0b35a)

Resolves issue #1053

DSCaskey committed:
add seamlyme welcome screen

ghost commented:
<details open>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=1057&src=https%3A%2F%2Fgithub.com%2FFashionFreedom%2FSeamly2D">Click on the image</a> for a new way to code review</b>
</summary>

####

[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/FashionFreedom/Seamly2D/1057/712e9327/3ec9ef5633139118b65b16cc8051c3d22994d7fc.svg)](https://app.codesee.io/r/reviews?pr=1057&src=https%3A%2F%2Fgithub.com%2FFashionFreedom%2FSeamly2D)

#### Legend
<img src="https://app.codesee.io/Legend_v3.png" alt="CodeSee Map legend" width="220" height="156">
</details>

csett86 committed:
Remove leftovers from splash screen

Dont open main window twice, revert to old processCommandLine logic

csett86 reviewed (approved):
Overall looks very good, I just did small fixups (reverting to the old immediate processing of the commandline args to avoid two seamly windows under macos) and removing the splash screen leftovers

csett86 commented:
mac build is failing due to a missing updated agreement in the Apple developer account:

> Error: HTTP status code: 403. A required agreement is missing or has expired. This request requires an in-effect agreement that has not been signed or has expired. Ensure your team has signed the necessary legal agreements and that they are not expired.

@slspencer could you please help us out here at https://developer.apple.com/account

![Bildschirmfoto 2024-01-06 um 22 05 33](https://github.com/FashionFreedom/Seamly2D/assets/1392875/64fea94c-b207-4eea-8908-82bb47f297eb)

DSCaskey committed:
remove qtimer call

Since the delay = 0, just call slot directly..

DSCaskey commented:
@csett86 

>reverting to the old immediate processing of the commandline args to avoid two seamly windows under macos

Since the delay = zero, we can just remove the QTimer call processCommandLine() directly. 

Also moved the "NOTing" of  showWelcome in the assignment. Makes it a bit clearer. If showWelcome was used more than once, would have to NOT every occurance.

DSCaskey committed:
revert remove qtimer call

For some reason without the QTimer call  even with a zero delay, is causing the seamlyME commandline tests to fail.

DSCaskey commented:
@csett86 

For some reason removing the QTimer call in main() is causing the SeamlyMe command line tests to fail??  Makes no sense, but just putting back the way it was.

csett86 commented:
I also did not follow through on the timer, just reverted it to the old value (same as you)
